### PR TITLE
Fixes #11759 and deletes expired keys

### LINKED ISSUE #11759: FileCache not expiring
Here is the sample code:

``` php
$di->setShared('cache', function() {
    //Cache data for one hour by default
    $frontCache = new \Phalcon\Cache\Frontend\Data(array(
        'lifetime' => 3600
    ));

    //File cache settings
    $cache = new \Phalcon\Cache\Backend\File($frontCache, array(
        'cacheDir' => __DIR__ . '/cache/'
    ));

    return $cache;
});
```

then 

``` php
$resetAfter = $rate['saved'] + $seconds - time();
$cache->save($cacheName, $rate, (float)$resetAfter);
```

After a while `$resetAfter` becomes negative, and `$cache->exists($cacheName)` still returns the value...

edit: not touched the file for > 3600 seconds, and the file is still there. Looks like a bug for me.

### PR DESCRIPTION
**Pull request by stratoss**
PR #11769 for 2.1.x branch

### CODE REVIEW

**Comment by Jurigag** [comment]
Can you squash commits ?

File: phalcon/cache/backend/file.zep [hunk 1]
```diff
@@ -116,33 +121,49 @@ class File extends Backend implements BackendInterface
...
+					} else {
+						let ttl = (int) lastLifetime;
+					}
 				}
 			} else {
 				let ttl = (int) lifetime;
 			}
 
 			clearstatcache(true, cacheFile);
-			let modifiedTime = (int) filemtime(cacheFile);
```
Comment (sergeyklay) [inline phalcon/cache/backend/file.zep]: in this case mowe `clearstatcache` down

Comment (stratoss) [inline phalcon/cache/backend/file.zep]: Will do

**Comment by stratoss** [comment]
I can't run the tests on my local machine for some reason... Will fix the tests and try to squash.

**Comment by stratoss** [comment]
Do you have any ideas why phpunit tests are passing, but SimpleCest and FileCest aren't ? It must be something 2.1.x related but couldn't figure it out...

**Comment by sergeyklay** [comment]
Closed in favor of https://github.com/phalcon/cphalcon/pull/13621. Thank you @stratoss!

**Pull request closed without merging**